src/AppBundle/Service/AwesomeSearch.php

```diff
@@ -1,70 +1,261 @@
+            $this->addPertinencePoint($typeResults, self::CONTENTPOINT);
+
+
+            $customersResults = $this->em->getRepository(SoftMain::class)->searchInCustomers($word);
+            $this->addPertinencePoint($customersResults, self::CONTENTPOINT);
+
+
+            $hostingCountryResults = $this->em->getRepository(SoftMain::class)->searchInHostingCountry($word);
+            $this->addPertinencePoint($hostingCountryResults, self::CONTENTPOINT);
+
+
+            $creationDateResults = $this->em->getRepository(SoftMain::class)->searchInCreationDate($word);
+            $this->addPertinencePoint($creationDateResults, self::CONTENTPOINT);
+
+            $webSiteResults = $this->em->getRepository(SoftMain::class)->searchInWebSite($word);
+            $this->addPertinencePoint($webSiteResults, self::CONTENTPOINT);
+
+            $knowledgeBaseLanguageResults = $this->em->getRepository(SoftMain::class)->searchInKnowledgeBaseLanguage($word);
+            $this->addPertinencePoint($knowledgeBaseLanguageResults, self::CONTENTPOINT);
+
 
-        foreach($words as $word){
+            $tagNameResults = $this->em->getRepository(SoftMain::class)->searchInTagName($word);
+            $this->addPertinencePoint($tagNameResults, self::TAGPOINT);
 
-            $nameResults = $this->em->getRepository(SoftMain::class)->searchInNames($word);
-            $descriptionResults = $this->em->getRepository(SoftMain::class)->searchInDescriptions($word);
-            $boolResults = $this->searchInYml($word);
 
+            $tagDescriptionResults = $this->em->getRepository(SoftMain::class)->searchInTagDescription($word);
+            $this->addPertinencePoint($tagDescriptionResults, self::TAGPOINT);
+
+            $this->rateByBool($word);
 
         }
 
-        return $finalResult;
+        $arrayToSort = $this->getFinalResult();
+
+        foreach ($arrayToSort as $key => $row) {
+            $point2[$key] = $row['points'];
+        }
+
+        array_multisort($point2, SORT_DESC, $arrayToSort);
+
+        $result = [];
 
+        foreach ( $arrayToSort as $cell ) {
+            $result[]  = $cell['soft'];
+        }
+
+
+        return $result;
     }
 
-    private function cleanQuery($query){
 
-        // Receive  a dirty query, give a clean array of words to explore
+    private function cleanQuery(string $query) :array
+    {
 
-        //explode
+        /**
+         *
+         * Receive the exact query ask by user, give a clean array of words to explore whithout stop words and other stuffs
+         *
+         */
+
+        $arrayOfWords = preg_split("/[\s,+\"'&%().]+/", $query);
+        $goodQuery = [];
+        $emptyWords = $this->getDatas()["EmptyWords"];
+        $arrayEmptyWords = explode(" ", $emptyWords);
+
+        foreach ($arrayOfWords as $word) {
+            $isDirtyOrNot = in_array($word, $arrayEmptyWords);
+            if ($isDirtyOrNot === false AND strlen($word)) {
```

Comment:
and strlen($word) : quelle est l'intention ?

Reply:
Lintention est : strlen($world) > 1

Reply:
fait